core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -253,15 +257,22 @@ public boolean delete(Supplier<JournalContext> journalContext, AlluxioURI uri) {
    */
   public String getMountPoint(AlluxioURI uri) throws InvalidPathException {
     String path = uri.getPath();
-
+    String candidatePath = ROOT;
     try (LockResource r = new LockResource(mReadLock)) {
       for (Map.Entry<String, MountInfo> entry : mState.getMountTable().entrySet()) {
         String alluxioPath = entry.getKey();
-        if (!alluxioPath.equals(ROOT) && PathUtils.hasPrefix(path, alluxioPath)) {
-          return alluxioPath;
+        // we choose a new candidate path if the previous candidatepath is a prefix
+        // of the current alluxioPath and the alluxioPath is a prefix of the path
+        if (!alluxioPath.equals(ROOT) && PathUtils.hasPrefix(path, alluxioPath)
```

Comment:
do we need to care about comparing with `ROOT` anymore?

Reply:
I was keeping it consistent with the previous implementation, but I think removing it makes it more clear.